mono/mini/aot-compiler.c

```diff
@@ -4327,13 +4327,19 @@ gboolean mono_aot_mode_is_full (MonoAotOptions *opts)
 	return opts->mode == MONO_AOT_MODE_FULL;
 }
 
+static gboolean
+mono_aot_mode_has_feature (MonoAotOptions *opts, MonoAotFeature feature)
+{
+	return opts->mode & feature;
+}
+
 static void add_generic_class_with_depth (MonoAotCompile *acfg, MonoClass *klass, int depth, const char *ref);
 
 static void
 add_generic_class (MonoAotCompile *acfg, MonoClass *klass, gboolean force, const char *ref)
 {
 	/* This might lead to a huge code blowup so only do it if neccesary */
-	if (!acfg->aot_opts.llvm_only && !mono_aot_mode_is_full (&acfg->aot_opts) && !force)
+	if (!acfg->aot_opts.llvm_only && !mono_aot_mode_has_feature (&acfg->aot_opts, MONO_AOT_FEATURE_GENERICCLASS) && !force)
```

Comment:
Would it be possible to add flags to AotCompile for these because this is very tedious to type.